### PR TITLE
Feature/sampling method

### PR DESCRIPTION
**Pull request by ameenhaq**
* Added Acceptance Test for Scala approach
* Added tests
* Added variables from frame data example

### CODE REVIEW

**Comment by willis7** [comment]
@ameenhaq, rather than commit the derby binaries, wouldn't it be more usable to provide migration scripts so a developer can provision the database from scratch? In most databases you can run an export as SQL, and then commit the SQL. This would then be supported by `git diff` and is much easier to review.

File: resources/features/Sampling.feature [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    a selection type parameter set as 'Census' or 'PrnSampling' (ommitting 'Universal') is then processed. We then use
+    the start point parameter (of each request), and select units in prn order until the required sample size (number)
+    is reached. All requests are outputted collectively as a DataFrame, with an appended 'Cell Number' to each to
+    indicate the source of request.
+
+    Scenario Outline: Sample size parameter is within bounds of the sorted records
+        Given a dataset and a stratification properties file containing sample size and start point parameter:
+            | data_input_path   | strat_properties        | expected_output   |
+            | <input_path>      | <strat_properties_path> | <output_path>     |
+        When the <language> sampling method is ran on the pre-filtered input
```
Comment (nigelhp) [inline resources/features/Sampling.feature]: "When a sample is created" ?

The business does not care that a "sampling method is ran".  They simply want a sample to be taken from a dataset.  Try to express the requirement not the implementation.

File: resources/features/Sampling.feature [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        Then a DataFrame of given sample size is returned
```
Comment (nigelhp) [inline resources/features/Sampling.feature]: Is there a global sample size parameter then?
I thought sampling parameters were applied on a cell by cell basis.
If one cell has the "census" sampling method, and another cell has the "prnSampling" method, then doesn't the size of the resulting sample depend on the size of the cell to which "census" was applied, in addition to the size of the sample requested from the "prnSampling" cell?

File: LICENSE [hunk 3]
```diff
@@ -1,21 +1,202 @@
...
-copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-SOFTWARE.
+Apache License
```
[the remark this answers is not in the document]
Comment (ameenhaq) [reply LICENSE]: This is not owned by Registers, best to speak to DAP-s (Owner) regarding this matter.

**Pull request closed without merging**